src/modules/compliance/src/lib/Procedure.cpp

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "Procedure.h"
+#include "Logging.h"
+
+#include <parson.h>
+#include <sstream>
+
+namespace compliance
+{
+    Procedure::~Procedure()
+    {
+        json_value_free(mAuditRule);
+        json_value_free(mRemediationRule);
+    }
+
+    Procedure::Procedure(Procedure&& other) noexcept
+        : mParameters(std::move(other.mParameters))
+        , mAuditRule(other.mAuditRule)
+        , mRemediationRule(other.mRemediationRule)
+    {
+        other.mAuditRule = nullptr;
+        other.mRemediationRule = nullptr;
+    }
+
+    Procedure& Procedure::operator=(Procedure&& other) noexcept
```

Comment:
This leaks existing resource of this object.
You don't need to provide this if you turn the members into unique_ptrs OR you need to swap out the members between `*this` and `other`.